### PR TITLE
Add guides on diagrams and collapse blocks

### LINKED ISSUE #65: Document collapse component and mermaid feature on mardown guide
Document collapse component and mermaid feature on mardown guide.

As a follow-up from the following review comments:
- _Originally posted by @talzid in https://github.com/thoth-tech/handbook/pull/57#discussion_r845774134_
- _Originally posted by @talzid in https://github.com/thoth-tech/handbook/pull/57#discussion_r845773881_

### PR DESCRIPTION
**Pull request by tancnle**
# Description

Add some information on usages of Mermaid diagrams and Collapse blocks in Markdown.

Fixes https://github.com/thoth-tech/handbook/issues/65

## Type of change

- [x] Documentation (update or new)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

File: docs/learning/training/markdown-guide.md [hunk 1]
```diff
@@ -63,28 +55,83 @@ Commonly used formatting options include the following:
...
+<details>
+<summary markdown="span">This is the summary text, click me to expand</summary>
+
+This is the detailed text.
+
+You can learn more about expected usage of this approach in the [ThothTech markdown guide](https://github.com/thoth-tech/handbook/blob/main/docs/learning/training/markdown-guide.md).
+
+</details>
+
+To add markdown in the collapsed heading, be sure to add:
```
Comment (ghost) [inline docs/learning/training/markdown-guide.md]: what do you mean by "add markdown in the collapsed heading"?  Maybe there is a different way to word it?

**Review by ghost — APPROVED** [review]
Looks good! Thank you for creating this Tan. 
I just left one comment cause I was confused by the wording.

File: docs/learning/training/markdown-guide.md (hunk 1 above)
Comment (tancnle) [reply docs/learning/training/markdown-guide.md]: Sorry for the confusion @talzid. I was under the impression that markdown content can't be rendered inside the `<details>` block but I am wrong. Apparently, the following text,

```markdown
<details>

### Manificient header
</details>
```
can be rendered correctly as 

<details>

### Manificient header
</details>

I will remove this line 👍🏼

**Review by kats8 — APPROVED** [review]
The Mermaid addition is cool, @tancnle Something new I have learned! :)

**Pull request merged**